tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/ConfidentialClientApplicationBuilderTests.cs

```diff
@@ -239,6 +239,24 @@ public void TestConstructor_WithCertificate_X509Certificate2()
             Assert.IsNotNull(cca.AppConfig.ClientCredentialCertificate);
         }
 
+        [TestMethod]
+        [DeploymentItem(@"Resources\valid_cert.cer")]
+        public void TestConstructor_WithCertificate_WithoutPrivateKey()
+        {
+            var cert = new X509Certificate2(
+                ResourceHelper.GetTestResourceRelativePath("valid_cert.cer"));
+
+            try
+            {
+                ConfidentialClientApplicationBuilder
+                      .Create(TestConstants.ClientId).WithCertificate(cert).Build();
+            } catch (MsalClientException e)
+            {
+                Assert.IsNotNull(e);
+                Assert.AreEqual(MsalError.CertWithoutPrivateKey, e.ErrorCode);
+            }
+        }
```

Comment:
If you use this style of testing, you need to add an `Assert.Fail` after block 257, otherwise if the `.WithCertificate` does not throw an exception, this test will still pass.

Reply:
might be better to assert that the exception is caught like some of the other tests do

Reply:
Added Assert.Fail